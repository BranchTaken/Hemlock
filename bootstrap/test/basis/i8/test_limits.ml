open! Basis.Rudiments
open! Basis
open I8

let test () =
  File.Fmt.stdout
  |> Fmt.fmt "bit_length="
  |> Uns.pp (bit_pop (bit_not zero))
  |> Fmt.fmt "\nmin_value="
  |> pp min_value
  |> Fmt.fmt " "
  |> fmt ~alt:true ~zpad:true ~width:2L ~base:Fmt.Hex ~pretty:true min_value
  |> Fmt.fmt "\nmax_value="
  |> pp max_value
  |> Fmt.fmt " "
  |> fmt ~alt:true ~zpad:true ~width:2L ~base:Fmt.Hex ~pretty:true max_value
  |> Fmt.fmt "\n"
  |> ignore

let _ = test ()
