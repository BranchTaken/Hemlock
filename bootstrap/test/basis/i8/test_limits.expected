bit_length=8
min_value=-128i8 -0x80i8
max_value=127i8 0x7fi8
